### PR TITLE
Customize Array.swapAt to pass down to the buffer.

### PR DESCRIPTION
**Pull request by natecook1000**
It looks like we aren't dropping the retain/release when using the default
implementation from MutableCollection; buffers do this swap with moves.

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please benchmark

**Comment by natecook1000** [comment]
@swift-ci Please benchmark

**Comment by natecook1000** [comment]
@swift-ci Please benchmark

**Comment by natecook1000** [comment]
@swift-ci Please benchmark

**Comment by gottesmm** [comment]
@natecook1000 can you show me the SIL for the rr you are talking about?

File: stdlib/public/core/Array.swift [hunk 1]
```diff
@@ -1311,6 +1311,12 @@ extension Array: RangeReplaceableCollection {
     }
     return _copyCollectionToContiguousArray(self)
   }
+  
+  @inlinable
+  public mutating func swapAt(_ i: Int, _ j: Int) {
+    _makeUniqueAndReserveCapacityIfNotUnique()
```
Comment (axiixc) [inline stdlib/public/core/Array.swift]: What is the cost of this call if `i == j`? You check for that in `ContiguousArrayBuffer.swapAt(_:_:)`, but at that point has the copy already been made?

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by natecook1000** [comment]
I don't think this is doing what I thought it was doing.

**Pull request closed without merging**